client/scripts/views/pages/view_proposal/linked_proposals_card.tsx

```diff
@@ -0,0 +1,159 @@
+    ]);
+  }
+}
+
+class ProposalSidebarLinkedSnapshot
+  implements
+    m.ClassComponent<{
+      proposal: OffchainThread;
+    }>
+{
+  private initialized: boolean;
+  private snapshot: SnapshotProposal;
+  private snapshotProposalsLoaded: boolean;
+  private space: SnapshotSpace;
+
+  view(vnode) {
+    const { proposal } = vnode.attrs;
+    if (!proposal.snapshotProposal) return;
+    if (!app.chain?.meta.chain.snapshot) return;
+
+    if (!this.initialized) {
+      this.initialized = true;
+
+      loadMultipleSpacesData(app.chain.meta.chain.snapshot).then((data) => {
+        for (const { space, proposals } of data) {
+          const matching_snapshot = proposals.find(
+            (sn) => sn.id === proposal.snapshotProposal
+          );
+          if (matching_snapshot) {
+            this.snapshot = matching_snapshot;
+            this.space = space;
+            break;
+          }
+        }
+        this.snapshotProposalsLoaded = true;
+        m.redraw();
+      });
+    }
+
+    let proposalLink = '';
+
+    if (this.space && this.snapshot) {
+      proposalLink = `${
+        app.isCustomDomain() ? '' : `/${proposal.chain}`
+      }/snapshot/${this.space.id}/${this.snapshot.id}`;
+    }
+
+    return link('a', proposalLink, [
+      `Snapshot: ${
+        !this.snapshotProposalsLoaded
+          ? proposal.snapshotProposal
+          : this.snapshot.title
+      }`,
+    ]);
+  }
+}
+
+export class LinkedProposalsCard
+  implements
+    m.ClassComponent<{
+      openStageEditor: () => void;
+      proposal: OffchainThread;
+      showAddProposalButton: boolean;
+    }>
+{
+  view(vnode) {
+    const { proposal, openStageEditor, showAddProposalButton } = vnode.attrs;
+
+    return (
+      <div class="LinkedProposalsCard">
+        {proposal.chainEntities.length > 0 ||
+        proposal.snapshotProposal?.length > 0 ? (
+          <h4>Proposals for Thread</h4>
+        ) : (
+          <h4>
+            {app.chain
+              ? 'Connect an on-chain proposal?'
+              : 'Track the progress of this thread?'}
+          </h4>
+        )}
+        <div class="links-container">
+          {proposal.chainEntities.length > 0 && (
+            <div class="proposal-chain-entities">
+              {proposal.chainEntities.map((chainEntity) => {
+                return (
+                  <ProposalSidebarLinkedChainEntity
+                    proposal={proposal}
+                    chainEntity={chainEntity}
+                  />
+                );
+              })}
+            </div>
+          )}
+          {proposal.snapshotProposal?.length > 0 && (
+            <div class="snapshot-container">
+              <ProposalSidebarLinkedSnapshot proposal={proposal} />
+            </div>
+          )}
+        </div>
+        {showAddProposalButton && (
+          <Button
+            rounded={true}
+            compact={true}
+            fluid={true}
+            label={app.chain ? 'Connect a proposal' : 'Update status'}
```

Comment:
there is only `app.chain` now! no need for this ternary if always true. "Connect a proposal" is fine

Reply:
we might do a PR where we find and remove all of these instances